ledger-service/lf-value-json/src/test/scala/com/digitalasset/daml/lf/value/json/ApiCodecCompressedSpec.scala

```diff
@@ -43,6 +43,79 @@ class ApiCodecCompressedSpec
   private def roundtrip(va: VA)(v: va.Inj[Cid]): Option[va.Inj[Cid]] =
     va.prj(jsValueToApiValue(apiValueToJsValue(va.inj(v)), va.t, typeLookup))
 
+  private object C /* based on navigator DamlConstants */ {
+    import shapeless.syntax.singleton._
+    val packageId0 = Ref.PackageId assertFromString "hash"
+    val moduleName0 = Ref.ModuleName assertFromString "Module"
+    def defRef(name: String) =
+      Ref.Identifier(
+        packageId0,
+        Ref.QualifiedName(moduleName0, Ref.DottedName assertFromString name))
+    val emptyRecordId = defRef("EmptyRecord")
+    val (emptyRecordDDT, emptyRecordT) = VA.record(emptyRecordId, RNil)
+    val simpleRecordId = defRef("SimpleRecord")
+    val simpleRecordVariantSpec = 'fA ->> VA.text :: 'fB ->> VA.int64 :: RNil
```

Comment:
What does `->>` do?

Reply:
It tags the right argument with the singleton type of the [literal] left argument; similarly to `@@` tags, its runtime value is just the right argument, though the compiler can naturally reify the tag back to a runtime value anywhere the literal is still apparent.  The `Record(k = v, ...)` is a macro doing something similar.